lib/pure/os.nim

```diff
@@ -60,8 +76,9 @@ proc osErrorMsg*(): string {.rtl, extern: "nos$1", deprecated.} =
                           nil, err, 0, addr(msgbuf), 0, nil) != 0'i32:
           result = $msgbuf
           if msgbuf != nil: localFree(msgbuf)
-  if errno != 0'i32:
-    result = $os.strerror(errno)
+  else:
```

Comment:
unless of course both getlasterror and errno should be checked - then both old & new code is wrong